### PR TITLE
Add fixed columns to uni-stark

### PR DESCRIPTION
**Pull request by Schaeff**

### CODE REVIEW

File: uni-stark/src/prover.rs [hunk 1]
```diff
@@ -156,6 +225,23 @@ where
...
+                    ))
+                    .chain(preprocessed_on_quotient_domain.iter().flat_map(
+                        |preprocessed_on_quotient_domain| {
+                            preprocessed_on_quotient_domain
+                                .vertically_packed_row(i_start + next_step)
+                        },
+                    ))
+                    .collect_vec(),
+                preprocessed_width,
+            );
```
Comment (georgwiese) [inline uni-stark/src/prover.rs:243]:
```suggestion
            let preprocessed = RowMajorMatrix::new(
                preprocessed_on_quotient_domain
                    .map(|on_quotient_domain| {
                        iter::empty()
                            .chain(on_quotient_domain.vertically_packed_row(i_start))
                            .chain(on_quotient_domain.vertically_packed_row(i_start + next_step))
                            .collect_vec()
                    })
                    .unwrap_or_default(),
                preprocessed_width,
            );
```

How about this?

File: uni-stark/src/verifier.rs [hunk 2]
```diff
@@ -21,6 +24,22 @@ pub fn verify<SC, A>(
...
+{
+    verify_with_key(config, None, air, challenger, proof, public_values)
+}
+
+#[instrument(skip_all)]
+pub fn verify_with_key<SC, A>(
+    config: &SC,
+    verifying_key: Option<&StarkVerifyingKey<SC>>,
+    air: &A,
+    challenger: &mut SC::Challenger,
```
Comment (georgwiese) [inline uni-stark/src/verifier.rs:39]: I think the challenger should observe the verification key as well!

File: uni-stark/src/prover.rs (hunk 1 above)
Comment (Schaeff) [reply uni-stark/src/prover.rs:243]: me gusta

File: uni-stark/src/verifier.rs (hunk 2 above)
Comment (Schaeff) [reply uni-stark/src/verifier.rs:39]: good point, I missed that!

**Review by georgwiese — APPROVED** [review]
LGTM

**Pull request merged**